collect_app/src/main/java/org/odk/collect/android/injection/config/AppDependencyModule.java

```diff
@@ -258,4 +259,9 @@ public CollectJobCreator providesCollectJobCreator() {
     public BackgroundWorkManager providesBackgroundWorkManager() {
         return new CollectBackgroundWorkManager();
     }
+
+    @Provides
+    public ConnectivityProvider providesConnectivityProvider() {
```

Comment:
Here I'd expect Dagger to provide the interface (`NetworkStateProvider`) rather that the implementation. For me the goal of that is that the objects that use it don't need to care about the implementation object at all - just the interface.

Reply:
Done.